chain/src/main/scala/org/bitcoins/chain/blockchain/ChainHandler.scala

```diff
@@ -43,17 +44,22 @@ case class ChainHandler(
   override def processHeader(header: BlockHeader)(
       implicit ec: ExecutionContext): Future[ChainHandler] = {
 
-    val blockchainUpdateF = Blockchain.connectTip(header, blockHeaderDAO)
+    val blockchainUpdateF = Blockchain.connectTip(header = header,
+                                                  blockHeaderDAO =
+                                                    blockHeaderDAO,
+                                                  blockchains = blockchains)
 
     val newHandlerF = blockchainUpdateF.flatMap {
-      case BlockchainUpdate.Successful(_, updatedHeader) =>
+      case BlockchainUpdate.Successful(newChain, updatedHeader) =>
         //now we have successfully connected the header, we need to insert
         //it into the database
         val createdF = blockHeaderDAO.create(updatedHeader)
         createdF.map { header =>
           logger.debug(
             s"Connected new header to blockchain, height=${header.height} hash=${header.hashBE}")
-          ChainHandler(blockHeaderDAO, chainConfig)
+
+          //what about old chain? The returned chain is _not_ necessarily the best chain I think
+          ChainHandler(blockHeaderDAO, chainConfig, Vector(newChain))
```

Comment:
Under what circumstances is this _not_ true?

Reply:
Our `ChainHandler` has a `Vector[Blockchain]`, while `BlockchainUpdate.Successful()` only returns _one_ chain. This is effectively discarding the competing chain from `ChainHandler`. There isn't even a guarantee here that the chain that got updated is the best chain.